pkg/core/store/badger/extension.go

```diff
@@ -0,0 +1,340 @@
+package badger
+
+import (
+	"context"
+	"time"
+
+	"github.com/fuseml/fuseml-core/pkg/domain"
+	"github.com/timshannon/badgerhold/v3"
+)
+
+// ExtensionStore is a wrapper around a badgerhold.Store that implements the domain.ExtensionStore interface.
+type ExtensionStore struct {
+	store *badgerhold.Store
+}
+
+// NewExtensionStore creates a new ExtensionStore.
+func NewExtensionStore(store *badgerhold.Store) *ExtensionStore {
+	return &ExtensionStore{store: store}
+}
+
+// AddExtension adds a new extension to the store.
+func (es *ExtensionStore) AddExtension(ctx context.Context, extension *domain.Extension) (*domain.Extension, error) {
+	extension.EnsureID(ctx, es)
+	extension.SetCreated(ctx)
+
+	err := es.store.Insert(extension.ID, extension)
+	if err != nil {
+		return nil, domain.NewErrExtensionExists(extension.ID)
```

Comment:
Just curious: can't there be a different reason the store operation fails?

This applies to all error checks after badgerstore operations

Reply:
According to the docs, insert only fails if a key already exists (see: https://github.com/timshannon/badgerhold#behavior-changes)

Reply:
Great